### ISSUE TITLE
Replace uses of `aria-` attributes with applicable `aria<x>Elements` properties

### ISSUE THREAD

**Issue by driskull** [Esri/calcite-components#13868]
**monday.com sync:** #11191891237

### Which Component

N/A

### Description

Wherever `aria-controls`, `aria-labelledby`, `aria-describedby` or other aria attributes that point to IDs of other elements are used, the newer `aria<X>Elements` property should be used.

For example, instead of `aria-controls`, the `ariaControlsElements` property should be used

### Proposed Advantages

The advantage is that unique IDs no longer need to be maintained and instead the elements themselves can be referenced.

Previously, I am seeing components use the `guid()` util along with a unique identifier to make sure that these ids are specific. That would no longer be necessary.

Another benefit is that these elements can be referenced across shadow DOM.

### Relevant Info

https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaControlsElements
https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaActiveDescendantElement
https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaDescribedByElements
https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaDetailsElements
https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaErrorMessageElements
https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaFlowToElements
https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaLabelledByElements

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components
